_ont_xgs/ont-fs-XGS-ONU-25-20NI.md

```diff
@@ -0,0 +1,586 @@
+- R4.4.20.016 (from FS.com docs)
+- R4.4.20.018 - 20230426061821
+- R4.4.20.022 - 20230612033149
+
+
+## List of partitions
+
+| dev    | size     | erasesize | name              |
+| ------ | -------- | --------- | ----------------- |
+| mtd0   | 00040000 | 00020000  | "ssb"             |
+| mtd1   | 00002000 | 00020000  | "uboot-env"       |
+| mtd2   | 00002000 | 00020000  | "dtb0"            |
+| mtd3   | 0003c000 | 00020000  | "kernel0"         |
+| mtd4   | 00300000 | 00020000  | "rootfs0"         |
+| mtd5   | 004c0000 | 00020000  | "dtb1"            |
+| mtd6   | 00300000 | 00020000  | "kernel1"         |
+| mtd7   | 004c0000 | 00020000  | "rootfs1"         |
+| mtd8   | 00001000 | 00020000  | "userdata"        |
+| mtd9   | 00001000 | 00020000  | "mfginfo1"        |
+| mtd10  | 00001000 | 00020000  | "mfginfo2"        |
+| mtd11  | 00001000 | 00020000  | "uboot-env2"      |
+
+This ONT supports dual boot. 
+
+`kernel0` and `rootfs0` respectively contain the kernel and firmware of the first image, `kernel1` and `rootfs1` the kernel and the firmware of the second one.
+
+
+# Usage
+
+## Login and Enable
+{% include alert.html content="This is an external file ([emulate ONT in QEMU](https://github.com/YuukiJapanTech/CA8271x)), so use it at your own risk!" alert="Note"  icon="svg-info" color="blue" %}
+
+This stick does not have a web console.
+To configure it, you must log in via `UART` or `Telnet`.
+
+| User | Password |
+| --- | --- |
+| `PON S/N` | `8digit HMAC-MD5 (PON S/N uppercase)` |
+
+To generate a password, emulate CIG ONT in Linux QEMU and run GponCLI and execute the following command.
+```
+sudo apt-get install qemu-user
+wget https://github.com/YuukiJapanTech/CA8271x/raw/main/emulate_CIG/GponCLI.tar.gz
+sudo tar -xvcf GponCLI.tar.gz -C /emulate_CIG
+ROOT=/emulate_CIG/root
+sudo chroot $ROOT /bin/GponCLI
+ONT> enable
+#ONT> system/debug/md5 "your PON-S/N uppercase" 8
+```
+
+For example, if the SN is `GPON2350004b`, the user name and password would be
+
+```
+#ONT> system/debug/md5 GPON2350004B 8
+MD5 Value: UzwugGYT
+```
+
+- User : `GPON2350004b`
+- Password : `UzwugGYT`
+
+UART does not ask for a login, it is possible to get a root shell without know the password.<br>
+
+{% include alert.html content="FS.com Stick XGS-ONU-25-20NI does not have PON S/N on the label. (The S/N field stamped on the label is FS.com internal S/N. it is not PON S/N), The only way to get the PON S/N on FS.com Stick is to check the boot Log from the UART!" alert="Note"  icon="svg-info" color="blue" %}
+
+## Root procedure
+After logging in via `telnet` or `UART`, you will first get the CLI with user privileges.
+
+```
+ONT>
+```
+
+The root CLI can be obtained by executing the `enable` command on this CLI.
+
+```
+ONT> enable
+#ONT>
+```
+
+can switch to the Linux root shell by executing the following command.
+
+```
+#ONT> system/shell
+#ONT/System/Shell> sh
+#
+```
+
+
+# GPON ONU status
```

Comment:
question: since the sdk is the same (saturn modified by cig) how different are the commands from https://hack-gpon.org/xgs/ont-nokia-xs-010x-q/ can you generalise it into one page and then import it into all pages?

Reply:
good. It should be the same including gpon and xgs, so I think there is no problem in standardizing
 (only register_id is buggy and does not work well...)

Reply:
Ok is a future things